Library/Homebrew/test/tap_spec.rb

```diff
@@ -192,7 +192,7 @@ def setup_completion(link:)
   end
 
   describe "#remote" do
-    it "returns the remote URL" do
+    it "returns the remote URL", :needs_network do
```

Comment:
`Tap#private?` ends up making a network call at the end of this block.

Reply:
Probably makes sense to scope this as `:needs_github_api`, even if it's just for self-documenting.

Reply:
I wonder how many other tests need the GitHub API right now. I'll add the tag and see if it can be applied to any other tests but it will just be for documentation. We don't even have a way to run tests by tag. Maybe that's something we should add but it honestly seems like it'd never get used.

Reply:
A lot of the `:needs_network` tests make network requests to Github. Though to be fair there aren't that many tests that need the network.

Reply:
I'd never run into the `RSpec/SortMetadata: Sort metadata alphabetically` cop before. Interesting...

Reply:
Addressed in https://github.com/Homebrew/brew/pull/16903/commits/836a26bc558ee930e18954c19b62cbdeb5f73997

Reply:
> Probably makes sense to scope this as `:needs_github_api`, even if it's just for self-documenting.

@reitermarkus @apainintheneck Sorry, I disagree here. Let's not add test tags like this just to be self-documenting when they don't actually do anything. It's confusing when all other tags do actually do things.

Reply:
`:needs_utils_curl` does the same as `:needs_network`.

I meant `:needs_github_api` to behave the same way, i.e. be an alias for `:needs_network`.

Reply:
> `:needs_utils_curl` does the same as `:needs_network`.

It doesn't. `:needs_network` results in those tests being relegated to `brew tests --online`. `:needs_utils_curl` does not.

`:needs_github_api` should do something different e.g. what `:needs_utils_curl` does but for the `utils/github` module or should be removed.

Reply:
Okay, then I agree with removing this again since `:needs_network` is required to even run these tests in the first place.

Reply:
Sounds good. I reverted that commit.